### ISSUE TITLE
Jbehave-junit-runner incorrectly shows scenarios being executed when mata keywords applied in Jbehave 4 version

### ISSUE THREAD

**Issue by falkoner** [valfirst/jbehave-junit-runner#85]
jbehave-core: 4.0.1
jbehave-junit-runner: 1.2.0
InetlliJ IDEA 14.1.4

Here is the piece of my Embedder:

```
@RunWith(JUnitReportingRunner.class)
public class ConsumerFrontEndStories extends JUnitStories {

public ConsumerFrontEndStories() {

configuredEmbedder().embedderControls().doGenerateViewAfterStories(true).doIgnoreFailureInStories(true)
                .doIgnoreFailureInView(false).useThreads(1).useStoryTimeoutInSecs(10800);
        configuredEmbedder().useMetaFilters(Arrays.asList("-skip", "+only"));
    }
```

Let's assume I have following story and want to run only second scenario in it:

```
A story is a collection of scenarios

Narrative:
In order to communicate effectively to the business some functionality
As a development team
I want to use Behaviour-Driven Development

Scenario:  A scenario three
Given step represents a precondition to an event
When step represents the occurrence of the event
Then step represents the outcome of the event

Scenario:  A scenario three
Meta:
@only
Given step represents a precondition to an event
When step represents the occurrence of the event
Then step represents the outcome of the event

Scenario:  A scenario three
Given step represents a precondition to an event
When step represents the occurrence of the event
Then step represents the outcome of the event
```

If I run this story in IntelliJ using JUnitReportingRunner as specified above, UI will show me that only **first** scenario is being executed, even though in fact **second** one is. That is in console I see all working fine but the _Run_ pane show that 1st story is executed.

If I remove @RunWith - all works fine though without nice output of JUnitReportingRunner. If I downgrade JBehave to 3.9.5 - all works as expected too.

**Comment by falkoner** [comment]
I guess we better downgrade to Jbehave core 3.10.0 or 3.9.5 meanwhile... Any ETA on this fix?

**Comment by britter** [comment]
@falkoner is this a problem with jbehave-junit-runner or with jjbehave itself?

**Comment by falkoner** [comment]
My understanding is that it's problem on jbehave-junit-runner side. Specifically in area of compatibility with Jbehave 4 branch. As result we downgraded Jbehave used in project to 3.10 from 4.0.1 and it works fine there now.

**Comment by britter** [comment]
Okay, I'll need time to investigate.

**Comment by falkoner** [comment]
@britter Any luck? Please let me know if I can help in any way. Though with update of IntelliJ to 15 (and the change in JUnit reporting results there) it all become way more broken for me.

**Comment by britter** [comment]
@falkoner honestly, I'm not working on this project anymore. If you need a fix, you'll have to supply a PR with accompanying test cases. Sorry!

**Comment by Guito** [comment]
Doing a  `currentScenario.getChildren().clear()` just before `notifier.fireTestFailure(new Failure(currentScenario, new AssertionError("scenario failed!")));` in JUnitSecenarioReporter.afterScenario makes the result look a bit better. It's not perfect but at least it's possible to work with it.

**Comment by valfirst** [comment]
@falkoner, @Guito, new version is released:
```xml
<dependency>
    <groupId>com.github.valfirst</groupId>
    <artifactId>jbehave-junit-runner</artifactId>
    <version>2.0.0</version>
</dependency>
```
It's based on JBehave 4.1.1. Have you had a chance to try it?

**Comment by cliviu** [comment]
@valfirst , is the new version already in mavencentral ?

**Comment by valfirst** [comment]
@cliviu , yes, it's already published, please, pay attention to the changed `groupId`: http://repo1.maven.org/maven2/com/github/valfirst/jbehave-junit-runner/2.0.0/

**Comment by cliviu** [comment]
ok, so there is also package name change...
Are there any known issues of the old version (1.2.0) with JBehave 4.1.1 ? We have a problem with the After methods of the lifecycle ... see also : https://github.com/serenity-bdd/serenity-core/issues/924
I will add this stacktrace also : 
[issue924.txt](https://github.com/valfirst/jbehave-junit-runner/files/1320965/issue924.txt)

**Comment by valfirst** [comment]
@cliviu , not sure if any matches your case, try to search: https://github.com/valfirst/jbehave-junit-runner/issues?utf8=%E2%9C%93&q=lifecycle. Is there any way to reproduce the issue without Serenity?

**Comment by cliviu** [comment]
ok, @valfirst, thanks, I will take a look
